src/node.cc

```diff
@@ -688,24 +615,6 @@ Local<Value> UVException(Isolate* isolate,
 }
 
 
-// Look up environment variable unless running as setuid root.
-bool SafeGetenv(const char* key, std::string* text) {
-#if !defined(__CloudABI__) && !defined(_WIN32)
-  if (linux_at_secure || getuid() != geteuid() || getgid() != getegid())
-    goto fail;
-#endif
-
-  if (const char* value = getenv(key)) {
-    *text = value;
-    return true;
-  }
-
-fail:
-  text->clear();
-  return false;
-}
```

Comment:
Per-process state, no point (and arguably wrong) into moving it into some per-environment abstraction.